src/extensions/extension-discovery.ts

```diff
@@ -197,23 +192,19 @@ export class ExtensionDiscovery {
     const extensionFolderName = path.basename(filePath);
 
     if (path.relative(this.localFolderPath, filePath) === extensionFolderName) {
-      const extensionName: string | undefined = Object
-        .entries(this.packagesJson.dependencies)
-        .find(([, extensionFolder]) => filePath === extensionFolder)?.[0];
+      const extension = Array.from(this.extensions.values()).find((extension) => path.dirname(extension.absolutePath) === path.dirname(filePath));
+
+      if (extension) {
+        const extensionName = extension.manifest.name;
 
-      if (extensionName !== undefined) {
         // If the extension is deleted manually while the application is running, also remove the symlink
         await this.removeSymlinkByPackageName(extensionName);
 
-        delete this.packagesJson.dependencies[extensionName];
-
-        // Reinstall dependencies to remove the extension from package.json
-        await this.installPackages();
-
         // The path to the manifest file is the lens extension id
         // Note that we need to use the symlinked path
-        const lensExtensionId = path.join(this.nodeModulesPath, extensionName, manifestFilename);
+        const lensExtensionId = extension.manifestPath;
```

Comment:
Can we actually remove this variable and use `extension.id`?

Reply:
Yes if they are the same thing.. I just tried to keep this compatible with current logic.

Reply:
They should be the same. In the old implementation the id was created manually because we didn't have the extension object itself.